### PR TITLE
Use E2E Tests for proper integration testing

### PR DESCRIPTION
**Pull request by roosnic1**
Needs to be updated with master and then added to the CI/CD

### CODE REVIEW

**Review by camilleory — APPROVED** [review]

**Review by penk85 — APPROVED** [review]

File: packages/api/src/server.ts [hunk 1]
```diff
@@ -95,7 +95,8 @@ export class WepublishServer {
     })
 
     const corsOptions = {
-      origin: '*',
+      origin: true,
```
Comment (bishoymelek) [inline packages/api/src/server.ts:98]: what does these do?

File: packages/editor/webpack.config.ts [hunk 2]
```diff
@@ -6,8 +6,8 @@ import {AssetListPlugin} from '@karma.run/webpack'
 import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin'
 import {CleanWebpackPlugin} from 'clean-webpack-plugin'
 
-export default (env: any, {mode}: any) =>
-  ({
+export default (env: any, {mode}: any) => {
```
Comment (bishoymelek) [inline packages/editor/webpack.config.ts:9]: do you want to get it back?

File: tests/e2e.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import * as process from "process";
+
+const loginName = Selector('input.username')
+const loginPassword = Selector('input.password')
+const createArticle = Selector('a').withAttribute('href', '/article/create')
+const metadataButton = Selector('button').child('i.rs-icon-newspaper-o')
+const createButton = Selector('button').child('i.rs-icon-save')
+const publishButton = Selector('button').child('i.rs-icon-cloud-upload')
+
+const closeButton = Selector('.rs-drawer-footer').child('button.rs-btn-primary')
```
Comment (bishoymelek) [inline tests/e2e.ts:12]: can we have it like `drawerCloseButton`? to differ it from modals for example?

File: packages/api/src/server.ts (hunk 1 above)
Comment (roosnic1) [reply packages/api/src/server.ts:98]: https://github.com/expressjs/cors#cors

File: packages/editor/src/client/app.tsx [hunk 4]
```diff
@@ -139,6 +141,18 @@ export function App() {
     setLang(currentLanguageMap.get(lng))
   })
 
+  const {session} = useContext(AuthContext)
+  const dispatch = useRouteDispatch()
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (roosnic1) [reply packages/editor/src/client/app.tsx:147]: We also check in the router but this is the place where we watch the `session` and the session is checked once you focus the browser tab with the editor again.

File: packages/editor/src/client/route.tsx [hunk 5]
```diff
@@ -250,7 +250,7 @@ export function RouteProvider({children}: RouteProviderProps) {
           } else {
             dispatch({
               type: RouteActionType.SetCurrentRoute,
-              route: next
+              route: {...next, hash: ''}
```
[the remark this answers is not in the document]
Comment (roosnic1) [reply packages/editor/src/client/route.tsx:253]: It's used to clear any URL hash but it's not needed. I removed it.

File: packages/editor/webpack.config.ts (hunk 2 above)
Comment (roosnic1) [reply packages/editor/webpack.config.ts:9]: yes

File: tests/e2e.ts (hunk 3 above)
Comment (roosnic1) [reply tests/e2e.ts:12]: This is only a test implementation. @camilleory has in here tests a better naming strategy.

**Review by bishoymelek — APPROVED** [review]
🎉

**Review by camilleory — APPROVED** [review]

**Pull request merged**